### ISSUE TITLE
Custom Envoy config

### ISSUE THREAD

**Issue by sgammon** [GoogleCloudPlatform/esp-v2#433]
Is it possible with ESPv2 to use a custom `envoy.yaml`, roughly as it was with ESPv1?

**Comment by nareddyt** [comment]
No, we do not support this currently. It's difficult since Envoy uses xDS API to receive the config from config manager. We have tried to support file-based configs in the past, but Envoy does not provide consistency guarantees.

Generally if there are small Envoy features you need, we can add a flag to turn them on. If it's a big change (like adding your own Envoy filter), we don't support it.

**Comment by qiwzhang** [comment]
The simplest way to add customization is to support  `--envoy_config_rewrite_url`.    If such url is provided,  ESPv2 will send its generated envoy config to this url,  its response is the modified envoy config.  That url server is external to ESPv2 container, it can be implemented in your http bakend or somewhere else.

**Comment by nareddyt** [comment]
Huh, that seems feasible. @sgammon do you have a concrete use-case that requires custom envoy config support? Or is this a theoretical question?

**Comment by sgammon** [comment]
@nareddyt [mod_pagespeed](https://github.com/apache/incubator-pagespeed-mod/issues/2024) is adding an envoy filter, and it would be awesome to unify proxy services for content and API proxying. that's my goal with this, but it seems like it may be pretty far off.
